bitnami/wavefront-prometheus-storage-adapter/templates/deployment.yaml

```
@@ -107,8 +121,22 @@ spec:
           resources: {{- toYaml .Values.resources | nindent 12 }}
           {{- end }}
           ports:
-            - containerPort: {{ .Values.containerPort }}
+            - containerPort: {{ coalesce .Values.containerPorts.http .Values.containerPort}}
               name: http
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.startupProbe.enabled }}
+          startupProbe:
+            httpGet:
+              path: /health
+              port: http
```

For the startupProbe, there is no need to check endpoint is healthy, simply check port is open:

```suggestion
           tcpSocket:
              port: http
```